### PR TITLE
[WebSub] Support non-JSON notification payload

### LINKED ISSUE #8458: [WebSub] Support/handle non-JSON notification payload
**Description:**
The current implementation of WebSub expects the payload content type to be JSON, for content delivery.

Other content-types to be supported need to be finalized and implemented.

### PR DESCRIPTION
**Pull request by MaryamZi**
## Purpose
The current implementation of WebSub expects the payload content type to be JSON, for content delivery.

This PR adds support for payload specified as string, xml, blob or ByteChannel in addition to json.

Fixes #8458

### CODE REVIEW

File: bvm/ballerina-core/src/main/java/org/ballerinalang/broker/BallerinaBrokerByteBuf.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package org.ballerinalang.broker;
```
Comment (anupama-pathirage) [inline bvm/ballerina-core/src/main/java/org/ballerinalang/broker/BallerinaBrokerByteBuf.java]: Missing license header

File: stdlib/ballerina-websub/src/main/ballerina/websub/commons.bal [hunk 2]
```diff
@@ -293,14 +295,150 @@ function validateSignature(string xHubSignature, string stringPayload, string se
...
-    json payload,
-    http:Request request,
+public type Notification object {
+
+    private {
+        http:Request request;
+    }
+
+    documentation {
+        Retrieves the query parameters of the content delivery request request, as a map.
```
Comment (anupama-pathirage) [inline stdlib/ballerina-websub/src/main/ballerina/websub/commons.bal]: duplicated word "request"?

File: stdlib/ballerina-websub/src/main/java/org/ballerinalang/net/websub/nativeimpl/ConstructBlob.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+import org.ballerinalang.natives.annotations.BallerinaFunction;
+import org.ballerinalang.natives.annotations.ReturnType;
+import org.ballerinalang.runtime.message.BlobDataSource;
+
+import java.io.IOException;
+
+import static org.ballerinalang.nativeimpl.io.IOConstants.BYTE_CHANNEL_NAME;
+
+/**
+ * Native function to add a subscription to the default Ballerina Hub's underlying broker.
```
Comment (anupama-pathirage) [inline stdlib/ballerina-websub/src/main/java/org/ballerinalang/net/websub/nativeimpl/ConstructBlob.java]: Is this description correct?

File: stdlib/ballerina-websub/src/main/java/org/ballerinalang/net/websub/nativeimpl/ConstructBlob.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+            try {
+                byte[] byteData = MimeUtil.getByteArray(channel.getInputStream());
+                channel.close();
+                context.setReturnValues(new BBlob((new BlobDataSource(byteData)).getValue()));
+            } catch (IOException e) {
+                context.setReturnValues(new BBlob(new byte[0]));
+            }
+        }
+    }
+
```
Comment (anupama-pathirage) [inline stdlib/ballerina-websub/src/main/java/org/ballerinalang/net/websub/nativeimpl/ConstructBlob.java]: Extra new line?

**Comment by MaryamZi** [comment]
@anupama-pathirage Addressed suggestions with 26cd20a

**Pull request merged**